sound/soc/soc-topology.c

```diff
@@ -1157,20 +1192,33 @@ static int soc_tplg_dapm_graph_elems_load(struct soc_tplg *tplg,
 			SNDRV_CTL_ELEM_ID_NAME_MAXLEN)
 			return -EINVAL;
 
```

Comment:
You still have a memory leak on returning -EINVAL, routes is not freed and stuff that was allocated but not used is lost as well since the route is not added yet.

Reply:
@plbossart @bardliao have fixed all opens in V4.